File: algebra-perm2a/js/ES_COMPUESTO.js
module.exports = (a, n, t, u) => {
  let x = Math.pow(a, u) % n;

  if (x == 1 || x == n - 1)
    return false;

  for (let i = 1; i == t; i++) {
    x = Math.pow(x, 2) % n;
    if (x == n - 1)
      return false;
  }

  return true;
}

File: algebra-perm2a/py/ES_COMPUESTO.py
import math
from POW_MOD import POW_MOD

def ES_COMPUESTO(a, n, t, u):
    x = POW_MOD(a, u, n)

    if (x == 1 or x == n - 1):
        return False

    for i in range(1,t,1):
        x = POW_MOD(x, 2, n)
        if (x == n - 1):
            return False

    return True
